python/dolfinx/fem/assemble.py

```diff
@@ -117,14 +131,14 @@ def create_matrix(a: Form, block_mode: typing.Optional[la.BlockMode] = None) ->
 # -- Scalar assembly ---------------------------------------------------------
 
 
-def assemble_scalar(M: Form, constants=None, coeffs=None):
+def assemble_scalar(M: Form, constants: typing.Optional[np.ndarray] = None, coeffs=None):
```

Comment:
Same typehint-comment about the np-array.
Should we also type-hint the coefficients, as done in pack_coefficients?